ucm2/Intel/avs/hdaudioB0D2/HDMI-3pb-HiFi.conf

```diff
@@ -0,0 +1,26 @@
+SectionDevice."HDMI1" {
+	Comment "HDMI/Display Port 1"
+
+	Value {
+		PlaybackPCM "hw:hdaudioB0D2,1"
+		JackDev "hdaudioB0D2 HDMI/DP,pcm=1"
```

Comment:
Not sure how the HDMI devices look for AVS. Are there also some controls to mute / unmute stream and set the stream parameters (for passthrough) ?

Reply:
We do not add any DSP controls. There are only codec ones, here is list from amixer:
# amixer -c4 controls
numid=1,iface=CARD,name='HDMI/DP,pcm=1 Jack'
numid=7,iface=CARD,name='HDMI/DP,pcm=2 Jack'
numid=13,iface=CARD,name='HDMI/DP,pcm=3 Jack'
numid=2,iface=MIXER,name='IEC958 Playback Con Mask'
numid=8,iface=MIXER,name='IEC958 Playback Con Mask',index=1
numid=14,iface=MIXER,name='IEC958 Playback Con Mask',index=2
numid=3,iface=MIXER,name='IEC958 Playback Pro Mask'
numid=9,iface=MIXER,name='IEC958 Playback Pro Mask',index=1
numid=15,iface=MIXER,name='IEC958 Playback Pro Mask',index=2
numid=4,iface=MIXER,name='IEC958 Playback Default'
numid=10,iface=MIXER,name='IEC958 Playback Default',index=1
numid=16,iface=MIXER,name='IEC958 Playback Default',index=2
numid=5,iface=MIXER,name='IEC958 Playback Switch'
numid=11,iface=MIXER,name='IEC958 Playback Switch',index=1
numid=17,iface=MIXER,name='IEC958 Playback Switch',index=2
numid=6,iface=PCM,name='ELD',device=1
numid=19,iface=PCM,name='Playback Channel Map',device=1
numid=12,iface=PCM,name='ELD',device=2
numid=20,iface=PCM,name='Playback Channel Map',device=2
numid=18,iface=PCM,name='ELD',device=3
numid=21,iface=PCM,name='Playback Channel Map',device=3

Should I use "IEC958 Playback Switch"? It seems to be enabled by default.

Reply:
It looks like that this standard macro should be used to handle those devices https://github.com/alsa-project/alsa-ucm-conf/blob/master/ucm2/codecs/hda/hdmi.conf . For usage, refer to https://github.com/alsa-project/alsa-ucm-conf/blob/master/ucm2/HDA/Hdmi.conf .

Reply:
So I'm trying to use this, on one of devices I have. There are 3 HDMI endpoints:
`card 4: hdaudioB0D2 [hdaudioB0D2], device 1: HDMI1 (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 4: hdaudioB0D2 [hdaudioB0D2], device 2: HDMI2 (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 4: hdaudioB0D2 [hdaudioB0D2], device 3: HDMI3 (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
`
and it seems like alsaucm sees something:
`# alsaucm -c hw:4 dump text
Verb.HiFi {
        Comment "Play HiFi quality Music"
        Device.HDMI1 {
                Comment "HDMI / DisplayPort 1 Output"
                Values {
                        CaptureCTL "_ucm0001.hw:hdaudioB0D2"
                        JackControl "HDMI/DP,pcm=3 Jack"
                        PlaybackCTL "_ucm0001.hw:hdaudioB0D2"
                        PlaybackPCM "_ucm0001.hw:hdaudioB0D2,3"
                        PlaybackPriority 500
                }
        }
}
`
However there is no HDMI endpoint in GUI at all?
![Screenshot from 2024-06-06 15-15-19](https://github.com/alsa-project/alsa-ucm-conf/assets/87762070/d0cfe517-a378-4827-97c0-434b2f85fe8f)

Here is how I set UCM files:
`# cat /usr/share/alsa/ucm2/conf.d/hdaudioB0D2/hdaudioB0D2.conf
Syntax 6

SectionUseCase."HiFi" {
        File "/Intel/avs/hdaudioB0D2/hdaudioB0D2-HiFi.conf"
        Comment "Play HiFi quality Music"
}
# cat /usr/share/alsa/ucm2/Intel/avs/hdaudioB0D2/hdaudioB0D2-HiFi.conf
Include.hdmi.File "/codecs/hda/hdmi.conf"

If.hdmi1 {
        Condition { Type AlwaysTrue }
        True.Macro.hdmi1.HDMI {
                Number 1
                Device 3
                Priority 500
        }
}

If.hdmi2 {
        Condition { Type AlwaysTrue }
        True.Macro.hdmi1.HDMI {
                Number 2
                Device 4
                Priority 600
        }
}

If.hdmi3 {
        Condition { Type AlwaysTrue }
        True.Macro.hdmi1.HDMI {
                Number 3
                Device 5
                Priority 700
        }
}
`
What am I missing?

Reply:
What's the jack state?

```
amixer -c hw: 4 cget iface=CARD,name='HDMI/DP,pcm=1 Jack'
```

If those values are false, the sound server do not list those devices (they are not available).

Reply:
So I've tried different machine just in case, but there is same behavior.
It sees device in controls, but nothing in GUI:
# amixer -c3 contents
numid=1,iface=CARD,name='HDMI/DP,pcm=1 Jack'
  ; type=BOOLEAN,access=r-------,values=1
  : values=on
numid=7,iface=CARD,name='HDMI/DP,pcm=2 Jack'
  ; type=BOOLEAN,access=r-------,values=1
  : values=off
numid=13,iface=CARD,name='HDMI/DP,pcm=3 Jack'
  ; type=BOOLEAN,access=r-------,values=1
  : values=off

Reply:
Is device number correct ? It seems that `Device` values should be in range 1,2,3 for your driver (see the aplay dump for PCM devices). All HDMI devices should be listed for alsaucm dump.

Reply:
Yes, after correcting the Device it appears to work now.